lib/ecto/query/builder.ex

```diff
@@ -357,10 +357,11 @@ defmodule Ecto.Query.Builder do
   def escape({comp_op, _, [left, right]} = expr, type, params_acc, vars, env)
       when comp_op in ~w(== != < > <= >=)a do
     assert_type!(expr, type, :boolean)
+    compare_str = Macro.to_string(expr)
```

Comment:
My only concern is that we are now storing strings in the generated artifact. And those can be large. What if we store only the opposite side, to reduce the amount of generated code?